### ISSUE TITLE
Describe how domain and challenge are sent to the holder

### ISSUE THREAD

**Issue by wyc** [decentralized-identity/presentation-exchange#29]
Add language for an optional `session` property which contains an object requiring at least a timestamp and hash of the entire Presentation Definition except for the hash using $existing_json_signing_standard. The object may include other fields such as a challenge/nonce, user ID, etc. to bind the response to a specific request. Presentation Definitions must have a present and comforming `session` property to be called Presentation Requests.

cc @OR13 and @dhh1188 who commented on #18 . 

I feel that "session" is kind of an overloaded word. Any suggestions for a better field name? @csuwildcat @mikelodder7 @glcohen

**Comment by wyc** [comment]
cc #23

**Comment by mikelodder7** [comment]
context_id?

**Comment by wyc** [comment]
is it okay that context_id holds an object?

**Comment by OR13** [comment]
can we link to examples or provide code snippets? I have no idea what we are talking about :)

**Comment by mikelodder7** [comment]
what kind of object would you want in the context? Also, I think context will clash with the name for JSON-LD folks.

**Comment by wyc** [comment]
We want to be able to specify something that looks like this:
```
"session": {
    "timestamp": "2020-06-25T23:20:50.52Z",
    "userId": "did:ethr:0x89205A3A3b2A69De6Dbf7f01ED13B2108B2c43e7",
    "challenge": "215632",
    "hash": "9a2d4c13922ab7313ef664684c4a31df5a22eae38f2a4124e739914d2e07ba38"
}
```

**Comment by mikelodder7** [comment]
I would move the userId outside of this. It's not clear what the hash is protecting and timestamp could replace challenge. To me, this looks unnecessary.

**Comment by wyc** [comment]
@mikelodder7 i guess we could also simplify things like this:
```
"challenge": "2020-06-25T23:20:50.52Z",
```

We were talking about allowing for an object so a requestor could create a Presentation Request for a specific user and channel, but I guess they can figure out how to put this into a challenge string. We could just recommend that the current time is incorporated into the challenge somehow, and provide some notes on setting a time window.

**Comment by mikelodder7** [comment]
If we can go to the millisecond precision that would better.

**Comment by glcohen** [comment]
The "session" here is for what purpose? To assert the validity of the request? Why would this not be handled at a higher layer (i.e. the JWT that wraps the presentation request)?

**Comment by wyc** [comment]
@glcohen It's not guaranteed that we would use JWT encoding for the Presentation Request or even VPs in the response.

More discussion here: https://github.com/decentralized-identity/presentation-exchange/pull/18#discussion_r438929433

**Comment by glcohen** [comment]
I see the concern if the verifier does not supply such a value; however, I do think it belongs at a different level and am worried about functionality leak into this proposal

**Comment by csuwildcat** [comment]
This should really happen outside the PE objects, somewhere in the envelop layer they are embedded in (i.e. OIDC, DIDComms, CHAPI, etc.)

**Comment by csuwildcat** [comment]
WG agreed to add spec notes on how nonces should be added to the outer envelop the PE objects are embedded within.

**Comment by OR13** [comment]
You can see how this works with the VP request spec today: https://w3c-ccg.github.io/vp-request-spec/#did-authentication-request... the verifier / issuer asks a wallet for a presentation like so:

```
{
    query: [ /* Presentation Request */],
    challenge: '99612b24-63d9-11ea-b99f-4f66f3e4f81a',
    domain: 'example.com'
  }
}
```

**Comment by csuwildcat** [comment]
Proposal: add note that this should be handled via use of these existing features in your outer envelope

**Comment by brentzundel** [comment]
This was addressed in #81
